dali/base/dafdesc.cpp

```diff
@@ -1702,6 +1733,30 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
         return *attr.get();
     }
 
+    virtual IPropertyTree *getHistory()
+    {
+        closePending();
+        if (history)
+            return history.getLink();
+        else
```

Comment:
I believe you've guaranteed that this cannot be true by now (i.e. all routes through constructors ensure)